src/App.tsx

```diff
@@ -1,60 +1,143 @@
 import './App.scss';
+import React, { useState } from 'react';
+// import React from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { FullTodo, Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+const prepareTodos = (
+  users: User[],
+  todos: Todo[],
+): FullTodo[] => {
+  return todos.map((todo) => ({
+    ...todo,
+    user: users.find((user) => user.id === todo.userId) || null,
+  }));
+};
+
+const todos = prepareTodos(usersFromServer, todosFromServer);
+const users = usersFromServer;
+
+export const App: React.FC = () => {
+  const [userFromSelect, setUser] = useState('0');
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [todosArr, setTodo] = useState(todos);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  };
+
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserError(false);
+    const { value } = event.target;
+
+    setUser(value);
+  };
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleError(false);
+    const { value } = event.target;
+
+    setTitle(value);
+  };
+
+  const handleAddTodo = () => {
+    const currentUser = users
```

Comment:
**suggestion**
i dont think you need handlers on both submit button and form. since they always trigger at the same time, you could merge `handleAddTodo`  and `handleSubmit` into one  handler